nptdms/writer.py

```diff
@@ -156,7 +217,7 @@ def _data_size(self):
 
     def _write_data(self, file):
         for obj in self.objects:
-            if hasattr(obj, 'data'):
+            if hasattr(obj, 'data') and not self.is_index_file:
```

Comment:
Rather than checking `is_index_file` here, I think it would be clearer to check this before calling `self._write_data` in the `write` method.